addon/components/base/bs-button.js

```diff
@@ -273,22 +336,20 @@ export default Component.extend(TypeClass, SizeClass, {
    * This will reset the state property to 'default', and with that the button's label to defaultText
    *
    * @method resetState
-   * @protected
+   * @private
    */
   resetState() {
-    this.set('textState', 'default');
+    this.set('state', 'default');
   },
 
   resetObserver: observer('reset', function() {
     if (this.get('reset')) {
-      scheduleOnce('actions', this, function() {
-        this.set('textState', 'default');
-      });
+      scheduleOnce('actions', this, 'resetState');
```

Comment:
Do you think we need test coverage that `state` is not set twice in a single runloop? Since there weren't any issues so far, triggering `reset` twice in a single run loop is not happening in real world applications.